### DISCUSSION TITLE
Installing Imagemagick version 7

### DISCUSSION THREAD

**Discussion by Bharad123** [ImageMagick/ImageMagick#4693] [Help] [answered]
Hello, I am trying to install latest version of imagemagick on our linux VM.

These are the steps I have followed so far.

$ git clone https://github.com/ImageMagick/ImageMagick.git ImageMagick-7.1.0
$ cd ImageMagick-7.1.0
$ ./configure
$ make

sudo make install
sudo ldconfig /usr/local/lib

it was installed with no errors. but seems like some delegates are missing. Whenever I try to run any identify or convert commands, 
I am getting the following error.

**no decode delegate for this image format png' @ error/constitute.c/readimage/572**

can someone help me with this please?

**Comment by fmw42** [comment, marked as answer]
Install from one of the binaries. Otherwise, you need to install all needed delegates, then install from source. You are just getting the IM source and no delegates the way you are installing.

https://imagemagick.org/script/download.php
https://imagemagick.org/script/install-source.php#unix

**Reply by Bharad123** [reply]
Thanks for the information.

**Reply by fmw42** [reply]
I am not an expert on Linux installation, so I will defer to one of the IM developers for further information.

**Reply by Bharad123** [reply]
> 

I've got it working now. Thanks for the information.
But I ran into another issue when dealing with TIFF files.
I am using Imagemagick version 6.9.10 and LibTiff version 4.0.3.
when i try to execute identify, i am getting the below error..

identify: Invalid TIFF directory; tags are not sorted in ascending order. `TIFFReadDirectoryCheckOrder' @ warning/tiff.c/TIFFWarnings/958.
identify: Unknown field with tag 444 (0x1bc) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/958.
identify: Incorrect count for "ProfileType"; tag ignored. `TIFFFetchNormalTag' @ error/tiff.c/TIFFErrors/609.
identify: Incorrect count for "GlobalParametersIFD"; tag ignored. `TIFFFetchNormalTag' @ error/tiff.c/TIFFErrors/609.

This image works fine for ImageMagick version 6.7.

**Reply by fmw42** [reply]
Please post your TIFF file. You will need to Zip it in order to drop it into this forum's window

**Reply by Bharad123** [reply]
[file.zip](https://github.com/ImageMagick/ImageMagick/files/7857388/file.zip)

Attached.

**Reply by fmw42** [reply]
Works fine for me on IM 6.9.12-34 and IM 7.1.0-19 on Mac OSX Sierra with LIBTIFF, Version 4.3.0.

I suggest you upgrade. Try just upgrading libtiff first.

You can get rid of the warnings (not the errors), by adding `-quiet` right after `convert` or `magick` depending upon your IM version.

**Reply by Bharad123** [reply]
Okay. we are using IM version 
Version: ImageMagick 6.9.10-68 Q16 x86_64 2021-09-22 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(3.1)
Delegates (built-in): bzlib cairo fontconfig freetype gslib jng jp2 jpeg lcms ltdl lzma openexr pangocairo png ps rsvg tiff wmf x xml zlib

and LibTiff:  
Loaded plugins: product-id, search-disabled-repos, subscription-manager, versionlock
Installed Packages
Name        : libtiff
Arch        : x86_64
Version     : 4.0.3
Release     : 35.el7
Size        : 483 k
Repo        : installed
From repo   : rhel-7-server-rpms
Summary     : Library of functions for manipulating TIFF format image files
URL         : http://www.remotesensing.org/libtiff/
License     : libtiff
Description : The libtiff package contains a library of functions for manipulating
            : TIFF (Tagged Image File Format) image format files.  TIFF is a widely
            : used file format for bitmapped images.  TIFF files usually end in the
            : .tif extension and they are often quite large.
            :
            : The libtiff package should be installed if you need to manipulate TIFF
            : format image files.

Ill try to update LibTiff and see how it behaves. Thanks for the suggestion.

**Reply by Bharad123** [reply]
> Works fine for me on IM 6.9.12-34 and IM 7.1.0-19 on Mac OSX Sierra with LIBTIFF, Version 4.3.0.
> 
> I suggest you upgrade. Try just upgrading libtiff first.
> 
> You can get rid of the warnings (not the errors), by adding `-quiet` right after `convert` or `magick` depending upon your IM version.

Hey, where can i get LibTiff V4.3.0. Is there any official site or any sort of URL that i can use?

**Reply by fmw42** [reply]
Have you tried Google?   See http://www.libtiff.org

**Reply by Bharad123** [reply]
Thank you.
